xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -309,10 +308,10 @@ std::shared_ptr<CPVRChannel> CPVRChannelGroup::GetByChannelEpgID(int iEpgID) con
   std::shared_ptr<CPVRChannel> retval;
   CSingleLock lock(m_critSection);
 
-  for (PVR_CHANNEL_GROUP_MEMBERS::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
+  for (const auto& memberPair : m_members)
```

Comment:
This changes the logic! Original code escapes fom the loop as soon as `retval` is non-zero!

I suggest to get rid of `retval`and early return directly from the loop on match.